services/configuration/src/lib.rs

```diff
@@ -75,32 +75,43 @@ extern crate exonum_testkit;
 #[cfg(test)]
 #[macro_use]
 extern crate pretty_assertions;
+extern crate serde_json;
+extern crate toml;
 
 pub use errors::ErrorCode;
-pub use schema::{MaybeVote, ProposeData, Schema, VotingDecision};
-pub use transactions::{ConfigurationTransactions, Propose, Vote, VoteAgainst};
-
+use exonum::encoding::serialize::json::reexport::Value;
 use exonum::{
     api::ServiceApiBuilder, blockchain::{self, Transaction, TransactionSet}, crypto::Hash,
-    encoding::Error as EncodingError, helpers::fabric::{self, Context}, messages::RawTransaction,
-    storage::Snapshot,
+    encoding::Error as EncodingError,
+    helpers::fabric::{self, keys, Command, CommandExtension, CommandName, Context},
+    messages::RawTransaction, storage::{Fork, Snapshot},
 };
+pub use schema::{MaybeVote, ProposeData, Schema, VotingDecision};
+use serde_json::to_value;
+pub use transactions::{ConfigurationTransactions, Propose, Vote, VoteAgainst};
 
 mod api;
+mod cmd;
+mod config;
 mod errors;
 mod schema;
 #[cfg(test)]
 mod tests;
 mod transactions;
 
+use cmd::{Finalize, GenerateCommonConfig, GenerateTestnet};
+use config::ConfigurationServiceConfig;
+
 /// Service identifier for the configuration service.
 pub const SERVICE_ID: u16 = 1;
 /// Configuration service name.
 pub const SERVICE_NAME: &str = "configuration";
 
-/// Configuration service.
+/// ConfigurationService config.
 #[derive(Debug, Default)]
-pub struct Service {}
+pub struct Service {
+    config: ConfigurationServiceConfig,
```

Comment:
We shouldn't save config inside `Service` structure, because we need to work within same config for all validators.

The only one valid way to track the same config for all validators - is looking in database.

Reply:
@ivan-ochc Any updates on this?

Reply:
@ilammy, as far as I understand such implementation is OK until ServiceFactory refactoring